third_party/amd/python/test/test_chained_dot_fp8.py

```diff
@@ -0,0 +1,194 @@
+        waves_per_eu = 2
+        num_warps = BLOCK_M // 32
+        num_stages = 1
+
+        grid = (triton.cdiv(q.shape[1], BLOCK_M), q.shape[0], 1)
+
+        _chained_dot[grid](
+            q,
+            k,
+            v,
+            o,
+            q_desc, k_desc, v_desc, s_sc, s_desc, o_sc,
+            q.stride(0),
+            q.stride(1),
+            q.stride(2),
+            k.stride(0),
+            k.stride(1),
+            k.stride(2),
+            v.stride(0),
+            v.stride(1),
+            v.stride(2),
+            o.stride(0),
+            o.stride(1),
+            o.stride(2),
+            Z=q.shape[0],
+            N=q.shape[1],
+            BLOCK_D=Lk,
+            BLOCK_M=BLOCK_M,
+            BLOCK_N=BLOCK_N,
+            USE_FP8=(q.dtype == float8),
+            waves_per_eu=waves_per_eu,
+            num_warps=num_warps,
+            num_stages=num_stages
+        )
+
+        return o
+
+
+chained_dot = chained_dot_fn.apply
+
+
+def to_float8(x, dtype=float8, margin: float = 1.0):
+    finfo = torch.finfo(dtype)
+    scale = finfo.max / x.abs().max().clamp(min=1e-12)
+    scale = math.pow(2, math.floor(math.log2(scale.float().item())) - margin)
+    x_scaled = (x.float() * scale).clamp(min=finfo.min, max=finfo.max)
+    return x_scaled.to(dtype), scale, 1.0 / scale
+
+
+@pytest.mark.parametrize('N, D, dtype',
+                         [(*shape, dtype)
+                          for shape in [(128, 32), (256, 128)]
+                          for dtype in ['fp8']])
+def test_chained_dot(N, D, dtype):
+    if dtype == 'fp8':
+        assert float8 is not None
+
+    BATCH = 1
+    q = torch.empty((BATCH, N, D), dtype=torch.float16, device="cuda").normal_(mean=0., std=0.5)
+    k = torch.empty((BATCH, N, D), dtype=torch.float16, device="cuda").normal_(mean=0., std=0.5)
+    v = torch.empty((BATCH, D, N), dtype=torch.float16, device="cuda").normal_(mean=0., std=0.5)
+
+    if dtype == 'fp8':
+        q_f8, _, q_desc = to_float8(q)
+        k_f8, _, k_desc = to_float8(k)
+        v_f8, _, v_desc = to_float8(v)
+
+        s = torch._scaled_mm(
+            q_f8[0], k_f8[0].transpose(0, 1),
+            out_dtype=torch.float32,
+            scale_a=torch.tensor(q_desc, dtype=torch.float32, device="cuda"),
+            scale_b=torch.tensor(k_desc, dtype=torch.float32, device="cuda"))
+        s_f8, s_sc, s_desc = to_float8(s)
+        ref = torch._scaled_mm(
+            s_f8, v_f8[0].transpose(0, 1),
+            out_dtype=torch.float32,
+            scale_a=torch.tensor(s_desc, dtype=torch.float32, device="cuda"),
+            scale_b=torch.tensor(v_desc, dtype=torch.float32, device="cuda"))
+        ref_f8, ref_sc, _ = to_float8(ref)
+
+        tri_out = chained_dot(
+            q_f8, k_f8, v_f8,
+            q_desc, k_desc, v_desc,
+            s_sc, s_desc, ref_sc)
+
+        assert tri_out.isnan().sum() == 0
+        torch.testing.assert_close(tri_out[0].float(), ref_f8.float(), atol=1e-2, rtol=3e-3)
```

Comment:
be careful with changing sensitivity (`atol=1e-2`). As the numbers are around 0 then you might miss numerical error with `atol=1e-2`